lib/dal/src/func/view.rs

```diff
@@ -1,4 +1,263 @@
-mod list;
+use serde::{Deserialize, Serialize};
+use telemetry::prelude::debug;
+use thiserror::Error;
 
-pub use list::FuncSummary;
-pub use list::FuncSummaryError;
+use crate::func::argument::{FuncArgument, FuncArgumentError, FuncArgumentId, FuncArgumentKind};
+use crate::func::authoring::{compile_langjs_types, compile_return_types};
+use crate::func::FuncKind;
+use crate::schema::variant::leaves::LeafInputLocation;
+use crate::{
+    ActionKind, ComponentId, DalContext, Func, FuncBackendKind, FuncBackendResponseType, FuncError,
+    FuncId, SchemaVariantError, SchemaVariantId,
+};
+
+pub mod summary;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum FuncViewError {
+    #[error("func error: {0}")]
+    Func(#[from] FuncError),
+    #[error("func argument error: {0}")]
+    FuncArgument(#[from] FuncArgumentError),
+    #[error("schema variant error: {0}")]
+    SchemaVariant(#[from] SchemaVariantError),
+}
+
+type FuncViewResult<T> = Result<T, FuncViewError>;
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributePrototypeView {}
+
+// #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+// #[serde(rename_all = "camelCase")]
+// pub struct ValidationPrototypeView {
+//     schema_variant_id: SchemaVariantId,
+//     prop_id: PropId,
+// }
+//
+#[remain::sorted]
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+#[serde(tag = "type", rename_all = "camelCase")]
+pub enum FuncAssociations {
```

Comment:
The home for these is in flux. The `authoring` and `view` modules share a lot of code. It is likely the module layout and API will change.